transport/http.go

```diff
@@ -144,30 +190,28 @@ func NewHTTPTransport() (*HTTPTransport, error) {
 			return verifyPeerCertificate(rawCerts, serverCert)
 		}
 	}
-
-	caCertPath := os.Getenv(envCACert)
-	if caCertPath != "" {
+	if opts.ServerCACert != "" {
 		rootCAs := x509.NewCertPool()
-		additionalCerts, err := ioutil.ReadFile(caCertPath)
+		additionalCerts, err := ioutil.ReadFile(opts.ServerCACert)
 		if err != nil {
-			return nil, errors.Wrapf(err, "failed to load root CA file from %s", caCertPath)
+			return nil, errors.Wrapf(err, "failed to load root CA file from %s", opts.ServerCACert)
 		}
 		if !rootCAs.AppendCertsFromPEM(additionalCerts) {
-			return nil, fmt.Errorf("failed to load CA certs from %s", caCertPath)
+			return nil, fmt.Errorf("failed to load CA certs from %s", opts.ServerCACert)
 		}
 		tlsConfig.RootCAs = rootCAs
 	}
 
 	client := &http.Client{
-		Timeout: serverTimeout,
+		Timeout: opts.ServerTimeout,
```

Comment:
Should NewHTTPTransportOptions return an error if ServerTimeout is invalid (negative)? I feel like it probably should, to avoid surprising errors at runtime.